File: Programmers/JAVA/lv1/lv1_모의고사.java
https://programmers.co.kr/learn/courses/30/lessons/42840

import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] score = new int[3];
        
        int[] stu1 = {1,2,3,4,5};
        int[] stu2 = {2,1,2,3,2,4,2,5};
        int[] stu3 = {3,3,1,1,2,2,4,4,5,5};

        for(int i = 0; i < answers.length; i++){
            if(stu1[i%stu1.length] == answers[i]) score[0]++;
            if(stu2[i%stu2.length] == answers[i]) score[1]++;
            if(stu3[i%stu3.length] == answers[i]) score[2]++;
        }

        int max = Math.max(Math.max(score[0],score[1]),score[2]);

        ArrayList<Integer> list = new ArrayList<Integer>();
        if(max == score[0]) list.add(1);
        if(max == score[1]) list.add(2);
        if(max == score[2]) list.add(3);

        int[] answer = new int[list.size()];

        for(int i = 0; i < list.size(); i++){
            answer[i] = list.get(i);
        }

        return answer;
    }
}

File: Programmers/C#/lv1/lv1_모의고사.cs
https://school.programmers.co.kr/learn/courses/30/lessons/42840

using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public int[] solution(int[] answers)
    {
        int[][] students = new int[3][];
        students[0] = new int[] { 1, 2, 3, 4, 5 };
        students[1] = new int[] { 2, 1, 2, 3, 2, 4, 2, 5 };
        students[2] = new int[] { 3, 3, 1, 1, 2, 2, 4, 4, 5, 5 };
        int[] score = new int[3] { 0, 0, 0 };

        for (int i = 0; i < answers.Length; i++)
        {
            for (int j = 0; j < score.Length; j++)
            {
                if (students[j][i % students[j].Length] == answers[i])
                    score[j]++;
            }
        }

        List<int> answer = new List<int>();
        for (int i = 0; i < score.Length; i++)
        {
            if (score[i] == score.Max())
                answer.Add(i + 1);
        }

        return answer.ToArray();
    }
}